src/v/cluster/partition_leaders_table.h

```diff
@@ -86,6 +91,7 @@ class partition_leaders_table {
     void update_partition_leader(
       const model::ntp&,
       model::revision_id,
+      model::initial_revision_id,
```

Comment:
I wonder if using `model::initial_revision_id` is correct here. AFAIR we introduced this to remember the revision of the topic that we restored from S3, which can be from a different cluster.

Reply:
the `initial_revision` changes every time the whole topic is deleted, and this is what we want to fence the leadership table updates with